enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/config/ConfigHolder.java

```diff
@@ -362,17 +366,56 @@ private void populateTMBinaryConfig(BinaryPublisher binary) {
 
     private void loadTrustStore() {
         try {
+
             trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
             trustStore.load(null);
-            String truststoreFilePath = getEnvVarConfig().getTrustedAdapterCertsPath();
-            TLSUtils.addCertsToTruststore(trustStore, truststoreFilePath);
+
+            // TODO: enable these with a config, got error when accessing configHolder
+            // properties since we call loadTrustStore method in ConfigHolder constructor
+            loadTrustedCertsToTrustStore();
+            loadDefaultCertsToTrustStore();
+
             trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
             trustManagerFactory.init(trustStore);
+
         } catch (KeyStoreException | CertificateException | NoSuchAlgorithmException | IOException e) {
             logger.error("Error in loading certs to the trust store.", e);
         }
     }
 
+    private void loadTrustedCertsToTrustStore() throws IOException {
+        String truststoreFilePath = getEnvVarConfig().getTrustedAdapterCertsPath();
+        TLSUtils.addCertsToTruststore(trustStore, truststoreFilePath);
```

Comment:
We need to address the following scenarios.

1. User provides a trusted certs via a volume mount.
2. User only use default certs from the trust manager.
3. Use above 1 and 2 options at the same time.

Can we achieve these scenarios?

Reply:
I think we need a config or at least an env config.

Reply:
added an env config for that. updated the PR description